content/hi/docs/tutorials/configuration/configure-java-microservice/configure-java-microservice.md

```diff
@@ -0,0 +1,68 @@
+---
+title: "माइक्रोप्रोफाइल, कॉन्फिगमैप्स और सीक्रेट्स का उपयोग करके कॉन्फिगरेशन को बाहरी बनाना"
+content_type: tutorial
+weight: 10
+---
+
+<!-- overview -->
+
+इस ट्यूटोरियल में आप सीखेंगे कि अपने माइक्रोसर्विस के कॉन्फ़िगरेशन को कैसे और क्यों इक्स्टर्नलाइज़ बनाना है।
+विशेष रूप से, आप सीखेंगे कि एनवायरनमेंट वेरिएबल सेट करने के लिए कुबेरनेट्स कॉन्फिगमैप्स और सीक्रेट्स का उपयोग कैसे करें और फिर माइक्रोप्रोफाइल कॉन्फिग का उपयोग करके उनका उपभोग करें।
+
+
+## {{% heading "prerequisites" %}}
+
+### कुबेरनेट्स कॉन्फ़िगमैप्स और सीक्रेट बनाना {#creating-kubernetes-configmaps-secrets}
+
+कुबेरनेट्स में डॉकर कंटेनर के लिए एनवायरनमेंट वेरिएबल सेट करने के कई तरीके हैं,
+इसमें शामिल हैं: Dockerfile, kubernetes.yml, Kubernetes configMaps, और Kubernetes Secrets.
+ट्यूटोरियल,में आप सीखेंगे कि अपने एनवायरनमेंट वेरिएबल सेट करने के लिए बाद वाले दो का उपयोग कैसे करें
+जिसके वैल्यू आपके माइक्रोसर्विसेज में इंजेक्ट किए जाएंगे। कॉन्फिगमैप्स का उपयोग करने के लाभों में से एक और
+रहस्य यह है कि इन्हें असाइन किए जाने सहित कई कंटेनरों में पुन: उपयोग किया जा सकता है
+विभिन्न कंटेनरों के लिए अलग-अलग एनवायरनमेंट वेरिएबल।
+
+कॉन्फिगमैप्स एपीआई ऑब्जेक्ट हैं जो गैर-गोपनीय कीय-वैल्यू पेयर को स्टोर करते हैं। इंटरैक्टिव में
+ट्यूटोरियल में आप सीखेंगे कि एप्लिकेशन के नाम को स्टोर करने के लिए कॉन्फिग मैप का उपयोग कैसे करें। अधिक जानकारी के लिए
+कॉन्फ़िगमैप्स के संबंध में जानकारी, आप दस्तावेज़ पा सकते हैं
+[यहां](/docs/tasks/configure-pod-container/configure-pod-configmap/)।
+
+हालाँकि सीक्रेट्स का उपयोग कीय-वैल्यू पेयर को स्टोर करने के लिए भी किया जाता है, लेकिन वे कॉन्फ़िगमैप्स से भिन्न होते हैं
+वे गोपनीय/संवेदनशील जानकारी के लिए अभिप्रेत हैं और बेस64 एन्कोडिंग का उपयोग करके स्टोर किए जाते हैं।
+यह रहस्यों को क्रेडेंशियल्स, कीय आदि जैसी चीज़ों को स्टोर करने के लिए जिसका उपयुक्त विकल्प बनाता है
+टोकन, जिनमें से पहला आप इंटरैक्टिव ट्यूटोरियल में करेंगे। अधिक जानकारी के लिए
+रहस्य, आप दस्तावेज़ [यहां](/docs/concepts/configuration/secret/) पा सकते हैं।
+
+
+### कोड से कॉन्फ़िग को इक्स्टर्नलाइज़ बनाना {# externalizing-config-from-code}
```

Comment:
There's a minor typo in this line where an extra space appears in the link text.